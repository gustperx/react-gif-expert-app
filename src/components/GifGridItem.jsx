import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="animate__animated animate__bounceIn">
      <img className="rounded-lg object-cover h-40" src={url} alt={title} />
      <p className="text-center mt-4 text-sm font-sans font-medium text-cyan-600">
        {title}
      </p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
