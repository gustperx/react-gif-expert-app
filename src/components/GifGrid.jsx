import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";

import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data, loaging } = useFetchGifs(category);

  return (
    <>
      <h3 className="text-2xl text-cyan-700 font-medium mb-4 animate__animated animate__fadeInDown">
        {category}
      </h3>

      {loaging && (
        <p className="text-2xl text-red-500 my-2 text-center font-semibold animate__animated animate__fadeInLeft">
          Loading...
        </p>
      )}

      <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>

      <hr className="my-8 font-semibold" />
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
