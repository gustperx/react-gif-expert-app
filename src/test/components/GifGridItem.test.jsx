import React from "react";
import { shallow } from "enzyme";
import { beforeEach, describe, expect, test } from "vitest";
import { GifGridItem } from "../../components/GifGridItem";

describe("Prueba en el GifGridItem", () => {
  let wrapper;

  const image = {
    title: "Demo Imagen",
    url: "https://google.com",
  };

  beforeEach(() => {
    wrapper = shallow(<GifGridItem {...image} />);
  });

  test("debe de hacer match con el snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener un parrafo con el titulo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(image.title);
  });

  test("debe de tener la imagen igual a la url y alt de los props", () => {
    const img = wrapper.find("img");

    //console.log(img.props());
    //console.log(img.prop("src"));

    expect(img.prop("src")).toBe(image.url);
    expect(img.prop("alt")).toBe(image.title);
  });

  test("debe de tener la clase animate__bounceIn", () => {
    const div = wrapper.find("div");

    //console.log(div.prop("className"));
    const className = div.prop("className");

    expect(className.includes("animate__bounceIn")).toBeTruthy();
  });
});
