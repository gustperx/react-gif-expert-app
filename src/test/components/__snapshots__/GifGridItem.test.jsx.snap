// Vitest Snapshot v1

exports[`Prueba en el GifGridItem > debe de hacer match con el snapshot 1`] = `
<div
  className="animate__animated animate__bounceIn"
>
  <img
    alt="Demo Imagen"
    className="rounded-lg object-cover h-40"
    src="https://google.com"
  />
  <p
    className="text-center mt-4 text-sm font-sans font-medium text-cyan-600"
  >
    Demo Imagen
  </p>
</div>
`;
